packages/components/src/components/post-megadropdown/post-megadropdown.tsx

```diff
@@ -104,9 +107,18 @@ export class PostMegadropdown {
   connectedCallback() {
     this.header = this.host.closest('post-header');
     if (this.header) {
-      this.header.addEventListener('postUpdateDevice', (event: CustomEvent<DEVICE_SIZE>) => {
-        this.device = event.detail;
-      });
+      this.header.addEventListener(
+        'postUpdateDevice',
+        (event: CustomEvent<DEVICE_SIZE>) => {
+          const newDevice = event.detail;
+          if (this.device !== newDevice) {
+            this.device = newDevice;
+            if (newDevice === 'desktop' && this.isVisible) {
+              this.animationClass = null;
+            }
+          }
+        },
+      );
```

Comment:
Just no!
Adding an eventlistener on the parent component is just not how web-components should communicate with each other, it's creating way too much and unnecessary dependencies!

Use the breakpoints utility to know if there is a breakpoint change or on which breakpoint you currently are.
Also, the utility already is optimize to only emit an event if there is a change, making this.device !== newDevice obsolete.